File: stack/python/nesting_score.py
def nesting_score(brackets: str) -> int:
    stack = [0]
    for ch in brackets:
        if ch == '[':
            stack.append(0)
        else:
            top = stack.pop()
            if top == 0:
                stack[-1] += 1
            else:
                stack[-1] += 2 * top
    return stack[0]

File: stack/csharp/NestingScore.cs
using System.Collections.Generic;

namespace StackSolutions;

public static class NestingScore
{
    public static int Score(string brackets)
    {
        var stack = new Stack<int>();
        stack.Push(0);
        foreach (var ch in brackets)
        {
            if (ch == '[')
            {
                stack.Push(0);
            }
            else
            {
                var top = stack.Pop();
                var value = top == 0 ? 1 : 2 * top;
                var updated = stack.Pop() + value;
                stack.Push(updated);
            }
        }

        return stack.Pop();
    }
}
